### PR TITLE
General cleanup and optimization of minor issues

### PR DESCRIPTION
**Pull request by GabrielRavier**
Numerous issues are fixed:
- Unmerged adjacent declarations and assignments (`anomaliesDetection`, `aws`)
- Unused values of certain variables (`costs`, `s3`, `users`)
- Redundant returns (`reports`)
- Bad usage of loops instead of just doing `append(a, b...)` (much simpler and very likely faster) (`cache`, `reports`, `routes`)
- Using `errors.New(fmt.Sprintf` instead of just `fmt.Errorf` (`aws`, `reports`, `users`)
- Using `fmt.Sprintf` when unnecessary (`aws`, `tagging`, `users`)
- Mispelled or missing type or func names in comments (`aws`, `costs`, `db`, `es`, `onDemandToRI`, `routes`, `tagging`, `usageReports`, `users`)
- Code that was way too verbose (`tagging.checkUserTagbotFreeTrial`)
- Using `time.Now().Sub` instead of `time.Since` (`routes`, `tagging`, `unusedAccounts`, `users`)
- Using `t.Add(time.Now())` instead of `time.Until(t)` (`unusedAccounts`)
- Unnecessary checks for `nil`, primarily w.r.t. guarding uses of `len` (`usageReports`, `users`, `util`)
- Useless use of `make(map[T]T, 0)` (often corrected to a more accurate size) (`aws`, `models`, `usageReports`)
- Unnecessary usage of `_` where no identifier is needed (`aws`, `costs`, `plugins`, `users`, `routes`)
- Useless usage of `break` (`errors`)
- Bad usage of type assertions that could be optimized (`errors`)
- Useless code that uses up time with no used results (`aws`)
- Fix some empty branches (`anomaliesDetection`, `server`)
- Usage of `+= 1` instead of `++` (`aws`, `costs`, `onDemandToRI`, `plugins`)
- Uncommented usage of blank imports that should probably be at least documented a bit (`db`, `plugins`)

### CODE REVIEW

**Comment by cruffinoni** [comment]
Adding the label "high priority". It fixes many bugs.

File: aws/routes/routeAwsStatus.go [hunk 1]
```diff
@@ -30,12 +30,15 @@ func getAwsAccountsStatus(r *http.Request, a routes.Arguments) (int, interface{}
 		l.Error("failed to get AWS accounts' bill repositories", err.Error())
 		return 500, errors.New("failed to retrieve bill repositories")
 	}
+	// Code unneeded for now
```
Comment (cruffinoni) [inline aws/routes/routeAwsStatus.go]: Can you explain why this code is no more needed, please?

File: errors/errors.go [hunk 2]
```diff
@@ -27,15 +27,15 @@ import (
 func GetErrorMessage(ctx context.Context, err error) error {
 	logger := jsonlog.LoggerFromContextOrDefault(ctx)
 	var formattedErr error
-	switch err.(type) {
+	switch err := err.(type) {
```
Comment (cruffinoni) [inline errors/errors.go:30]: This modification is debatable

File: s3/costs/prepare_response.go [hunk 3]
```diff
@@ -108,7 +108,7 @@ func parseBuckets(buckets BucketsInfo, parsedDocument bucket, resultType string)
 		if isValidBucket(bucketName) {
 			bucketInfo := getBucketInfoByName(buckets, bucketName)
```
Comment (cruffinoni) [inline s3/costs/prepare_response.go]: `bucketInfo` could be created directly after the condition below, couldn't it?

File: aws/routes/routeAwsStatus.go (hunk 1 above)
Comment (GabrielRavier) [reply aws/routes/routeAwsStatus.go]: Made the comment clearer.

Btw do tell if you think the code should simply be removed, I kept it (though commented) in the case it was there to be used in some later functionality.

File: errors/errors.go (hunk 2 above)
Comment (GabrielRavier) [reply errors/errors.go:30]: How, exactly ? It seems to me like it simplifies the code nicely (it's probably also a minor optimization now that I think about it) compared to manual type assertions for every case... If you want, I can change the variable name if you find that it makes things unclear

File: s3/costs/prepare_response.go (hunk 3 above)
Comment (GabrielRavier) [reply s3/costs/prepare_response.go]: Done.

File: errors/errors.go (hunk 2 above)
Comment (cruffinoni) [reply errors/errors.go:30]: After a second look, I do agree with your changes. My bad.

File: aws/routes/routeAwsStatus.go (hunk 1 above)
Comment (cruffinoni) [reply aws/routes/routeAwsStatus.go]: It is better to explain directly in the code why this is not necessary at the moment. We will keep it, however. You never know.

**Review by cruffinoni — APPROVED** [review]
Great, looks good. We will wait for @simonmeyerrr or someone else to review it, as it's a huge PR.

**Review by simonmeyerrr — APPROVED** [review]

**Pull request merged**